maestro-cli/src/main/java/maestro/cli/analytics/Analytics.kt

```diff
@@ -113,49 +113,55 @@ object Analytics {
-            return
-        }
+            if (!analyticsState.enabled) {
+                logger.trace("Analytics disabled with config file, not uploading")
+                return
+            }
 
-        if (!uploadConditionsMet) {
-            logger.trace("Upload conditions not met, not uploading")
-            return
-        }
+            if (!uploadConditionsMet) {
+                logger.trace("Upload conditions not met, not uploading")
+                return
+            }
 
-        val report = AnalyticsReport(
-            uuid = analyticsState.uuid,
-            freshInstall = !hasRunBefore,
-            cliVersion = EnvUtils.CLI_VERSION?.toString() ?: "Unknown",
-            os = EnvUtils.OS_NAME,
-            osArch = EnvUtils.OS_ARCH,
-            osVersion = EnvUtils.OS_VERSION,
-            javaVersion = EnvUtils.getJavaVersion().toString(),
-            xcodeVersion = IOSEnvUtils.xcodeVersion,
-            flutterVersion = EnvUtils.getFlutterVersionAndChannel().first,
-            flutterChannel = EnvUtils.getFlutterVersionAndChannel().second,
-            androidVersions = AndroidEnvUtils.androidEmulatorSdkVersions,
-            iosVersions = IOSEnvUtils.simulatorRuntimes,
-        )
+            val report = AnalyticsReport(
+                uuid = analyticsState.uuid,
+                freshInstall = !hasRunBefore,
+                cliVersion = EnvUtils.CLI_VERSION?.toString() ?: "Unknown",
+                os = EnvUtils.OS_NAME,
+                osArch = EnvUtils.OS_ARCH,
+                osVersion = EnvUtils.OS_VERSION,
+                javaVersion = EnvUtils.getJavaVersion().toString(),
+                xcodeVersion = IOSEnvUtils.xcodeVersion,
+                flutterVersion = EnvUtils.getFlutterVersionAndChannel().first,
+                flutterChannel = EnvUtils.getFlutterVersionAndChannel().second,
+                androidVersions = AndroidEnvUtils.androidEmulatorSdkVersions,
+                iosVersions = IOSEnvUtils.simulatorRuntimes,
+            )
 
-        logger.trace("Will upload analytics report")
-        logger.trace(report.toString())
+            logger.trace("Will upload analytics report")
+            logger.trace(report.toString())
 
-        try {
             ApiClient(EnvUtils.BASE_API_URL).sendAnalyticsReport(report)
             updateAnalyticsState()
         } catch (e: ConnectException) {
-            // This is fine. We don't care that much about analytics to bug user about it.
+            // This is fine. The user probably doesn't have internet connection.
+            // We don't care that much about analytics to bug user about it.
             return
+        } catch (e: Exception) {
+            // This is less fine. Don't crash, but ask the user to report this.
+            println("Exception ocurred while uploading analytics. This is not a fatal issue, but we ask you to report it.")
+            println(e.message)
+            println(e.stackTraceToString())
```

Comment:
Why do we want to print it on cli?

I don't think this would have any meaning for user. For us yes then maybe log this in maestro.log or some log file instead.

Reply:
We want to make sure analytics data is flowing to us. If we will just swallow the exception, we will never learn if we silently break sth in analytics.

Reply:
I don't really see value of showing this to user. Maybe there is a better option to provide this observability then?

Users don't really care about analytics, we do. So, maybe we opt another way for this observability WDYT?

Reply:
~~I think the best solution would be to connect Sentry (or any other log reporting service)?~~ That would be a lot of work though.

Actually, we already have an `ErrorReporter`. I just took a look at BigQuery, and it looks pretty solid. ([query link](https://console.cloud.google.com/bigquery?ws=!1m7!1m6!12m5!1m3!1sperf-dev-289002!2sus-west1!3s750c7a13-e1c7-422d-9d27-f4507a1faf3f!2e1) - internal only)

The problem is that [we don't wrap our whole app with the ErrorReporter](https://github.com/mobile-dev-inc/maestro/blob/9e0f64257ec576022a85829cb5d6a5b2d03ea046/maestro-cli/src/main/java/maestro/cli/App.kt#L106), so not all errors are reported. This is a separate issue though.

For now I agree with you, let's remove this message for the good of our users.

Reply:
Yes you can create a separate issue for this. Its definitely useful though!